src/mono/mono/metadata/marshal.c

```diff
@@ -3161,8 +3223,9 @@ mono_emit_marshal (EmitMarshalContext *m, int argnum, MonoType *t,
 	if (!m->runtime_marshalling_enabled)
 		return mono_emit_disabled_marshal (m, argnum, t, spec, conv_arg, conv_arg_type, action);
 
-	return mono_emit_marshal_ilgen(m, argnum, t, spec, conv_arg, conv_arg_type, action, get_marshal_cb());
-}
+	mono_component_marshal_ilgen()->install_callbacks_mono(mono_marshal_get_mono_callbacks_for_ilgen());
```

Comment:
Why is this the place to install the callbacks back into mono?

Reply:
This is the last place before it is used, but I guess it makes more sense to initialize it just after the component is loaded.